sdk/python/tests/integration/offline_store/test_universal_historical_retrieval.py

```diff
@@ -0,0 +1,357 @@
+        customer_df.to_dict("records"), customer_fv.batch_source.event_timestamp_column
+    )
+
+    # Manually do point-in-time join of orders to drivers and customers records
+    for order_record in order_records:
+        driver_record = find_asof_record(
+            driver_records,
+            ts_key=driver_fv.batch_source.event_timestamp_column,
+            ts_start=order_record[event_timestamp] - driver_fv.ttl,
+            ts_end=order_record[event_timestamp],
+            filter_key="driver_id",
+            filter_value=order_record["driver_id"],
+        )
+        customer_record = find_asof_record(
+            customer_records,
+            ts_key=customer_fv.batch_source.event_timestamp_column,
+            ts_start=order_record[event_timestamp] - customer_fv.ttl,
+            ts_end=order_record[event_timestamp],
+            filter_key="customer_id",
+            filter_value=order_record["customer_id"],
+        )
+
+        order_record.update(
+            {
+                (f"driver_stats__{k}" if full_feature_names else k): driver_record.get(
+                    k, None
+                )
+                for k in ("conv_rate", "avg_daily_trips")
+            }
+        )
+
+        order_record.update(
+            {
+                (
+                    f"customer_profile__{k}" if full_feature_names else k
+                ): customer_record.get(k, None)
+                for k in (
+                    "current_balance",
+                    "avg_passenger_count",
+                    "lifetime_trip_count",
+                )
+            }
+        )
+
+    # Convert records back to pandas dataframe
+    expected_df = pd.DataFrame(order_records)
+
+    # Move "event_timestamp" column to front
+    current_cols = expected_df.columns.tolist()
+    current_cols.remove(event_timestamp)
+    expected_df = expected_df[[event_timestamp] + current_cols]
+
+    # Cast some columns to expected types, since we lose information when converting pandas DFs into Python objects.
+    if full_feature_names:
+        expected_column_types = {
+            "order_is_success": "int32",
+            "driver_stats__conv_rate": "float32",
+            "customer_profile__current_balance": "float32",
+            "customer_profile__avg_passenger_count": "float32",
+        }
+    else:
+        expected_column_types = {
+            "order_is_success": "int32",
+            "conv_rate": "float32",
+            "current_balance": "float32",
+            "avg_passenger_count": "float32",
+        }
+
+    for col, typ in expected_column_types.items():
+        expected_df[col] = expected_df[col].astype(typ)
+
+    return expected_df
+
+
+def stage_orders_bigquery(df, table_id):
+    client = bigquery.Client()
+    job_config = bigquery.LoadJobConfig()
+    df.reset_index(drop=True, inplace=True)
+    job = client.load_table_from_dataframe(df, table_id, job_config=job_config)
+    job.result()
+
+
+class BigQueryDataSet:
```

Comment:
seems like a weird spot to have this? i.e. in what seems like an offline store agnostic test

Reply:
Yeah this is copy-pasta from the original test which had a bunch of fixtures in the same file. I'm going to kill em all